ROADMAP.md

```diff
@@ -0,0 +1,359 @@
+  - [5. Testground Is Usable by Non-Testground experts](#5-testground-is-usable-by-non-testground-experts)
+  - [6. Support libp2p's interoperability testing story and ProbeLabs work as a way to drive "critical" Testground improvements](#6-support-libp2ps-interoperability-testing-story-and-probelabs-work-as-a-way-to-drive-critical-testground-improvements)
+- [Appendix: Problems we focus on](#appendix-problems-we-focus-on)
+
+## Context
+
+> Testground is a platform for testing, benchmarking, and simulating distributed and peer-to-peer systems at scale.
+> It's designed to be multi-lingual and runtime-agnostic, scaling gracefully from 2 to 10k instances when needed.
+
+Testground was used successfully at Protocol Labs to validate improvements like libp2p's gossipsub security extensions,
+IPFS' massive DHT and Bitswap improvement, and Filecoin improvements. Today, we are actively working on it to support libp2p interoperability testing.
+
+### About the Roadmap
+
+This document consists of two sections:
+
+- [Milestones](#milestones) is the list of key improvements we intend to release with dates and deliverables. It is the "deliverable" oriented slice of our plan.
+- [Appendix: Problems we focus on](#appendix-problems-we-focus-on) is the list of areas of improvement we intend to focus on. Some contain a few examples of milestones. It is the "problem" oriented slice of our plan and might cover multiple releases and multiple "milestones".
+
+There is an issue dedicated to [Tracking Discussions around this Roadmap](https://github.com/testground/testground/issues/1491).
+
+The timeline we share is our best-educated guess (not a hard commitment) around when we plan to provide critical improvements.
+
+Where possible, we've shared a list of deliverables for every Milestone. These lists are not exhaustive or definitive; they aim to create a sense of "what are concrete outcomes" delivered by an improvement.
+
+As we agree on this Roadmap and refine it, we'll create & organize missing EPICs.
+
+Making Testground project management sustainable is [one of our key milestones](#3-improve-our-development--testing-infrastructure-to-meet-engineering-standards).
+
+## Vision
+
+_This section is still a very high-level draft_
+
+Testground As A Service embodies our long-term vision.
+
+- A single, scalable, installation that one or more organizations can use in all their projects,
+- The ability to experiment with large-scale networks and simplify the integration testing of "any" network-related code,
+- The ability to track the impact of a change in terms of stability & performance across multiple projects,
+  - An example: having the ability to run IPFS benchmarks and simulations with different combinations of libraries. This would help us measure regression and improvements as soon as they occur in upstream dependencies.
```

Comment:
I don't quite get this example. Isn't it already kind of solved? A new library is released. Dependabot makes a PR to Kubo to upgrade the dependency. We run a benchmark to see how it's doing comparing to release, master, etc. on the PR.

When I read the line above the example, the first thing I thought about was testing in the opposite direction. I make a change in a library, how is it going to affect my users.

Side note: Do we want to/are we already having a discussion on what kind of testing testground should be focused on? Correctness, performance, etc? I imagine at some point we'd want to have a list of testing tools (testground, thunderdome, etc.) and the usecases they are the best at.

Reply:
(let's talk about this)

> Side note: Do we want to/are we already having a discussion on what kind of testing testground should be focused on? Correctness, performance, etc? I imagine at some point we'd want to have a list of testing tools (testground, thunderdome, etc.) and the usecases they are the best at.

That deserves a whole discussion with the Thunderdome team.